cmake/AddGncDocTargets.cmake

```diff
@@ -1,47 +1,68 @@
+#
+# A common function to add a target.
+#
+# FUNCTION:
+#   add_gnc_doc_targets
+# ARGUMENTS:
+# - docname: The basename of the main xml file. Will be used to locate
+#            this primary xml file and for various output files/directories.
+#            Either "gnucash-guide" or "gnucash-help" now.
+# - entities: A list of all xml files this document is composed of. ONLY filename, WITHOUT path.
+#             It does NOT contain "${docname}.xml".
+#
 function (add_gnc_doc_targets docname entities)
 
     get_filename_component(lang ${CMAKE_CURRENT_SOURCE_DIR} NAME)
 
     file(GLOB_RECURSE figures
         "${CMAKE_CURRENT_SOURCE_DIR}/figures/*.png"
-        "${CMAKE_CURRENT_SOURCE_DIR}/figures/figures/*.svg")
+        "${CMAKE_CURRENT_SOURCE_DIR}/figures/*.svg")
+
+    # After CMake 3.12,
+    # can be replaced with list(TRANSFORM <list> <ACTION> [...])
+    file(GLOB_RECURSE figures_dist
+        RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+        figures/*.png figures/*.svg)
+
+    set(dtd_files "${CMAKE_SOURCE_DIR}/docbook/gnc-docbookx.dtd"
+                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-C.dtd"
+                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-${lang}.dtd")
+    list(REMOVE_DUPLICATES dtd_files)
+    list(APPEND source_files ${dtd_files})
```

Comment:
Variable source_files is not used at this level and reset in the functions we call later, so it can be removed here.